lib/ansible/plugins/lookup/filament_lookup.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: filament_lookup
+    author: Ansible core team
+    version_added: "2.7"
+    short_description: reuturn output of process table.
+    description:
+      - If recieved no arguments, return the whole process table. If recieved arguments, search the process tale with the argument as a process name.
+"""
+
+EXAMPLES = """
+  result: "{{lookup('filament_lookup')}}"
+  tasks:
+  - debug:
+    msg: "result is {{result}}"
+"""
+
+RETURN = """
+content of process table
+"""
+
+
+from ansible.plugins.lookup import LookupBase
+
+try:
+    import subprocess
+    from ansible.errors import AnsibleError
+    from ansible.module_utils._text import to_text
+    import epdb
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+def run_command(terms):
```

Comment:
You could probably use argument unpacking if you tend to accept multiple args:
```python
def run_command(*terms):
    command = 'ps aux'
    if terms:
        command = '%s | grep %s' % (command, ' '.join(terms))

```

https://docs.python.org/3/tutorial/controlflow.html#more-on-defining-functions

Also common way of having optional function args is to default them to `None`:
```python
def run_command(terms=None):
    if terms is None:
        terms = []  # have some reasonable default if you need
```